components/scream/src/physics/spa/spa_functions.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef SPA_FUNCTIONS_HPP
+#define SPA_FUNCTIONS_HPP
+
+#include "share/scream_types.hpp"
+
+#include "ekat/ekat_pack_kokkos.hpp"
+#include "ekat/ekat_workspace.hpp"
+
+namespace scream {
+namespace spa {
+
+template <typename ScalarT, typename DeviceT>
+struct SPAFunctions
+{
+
+  //
+  // ------- Types --------
+  //
+
+  using Scalar = ScalarT;
+  using Device = DeviceT;
+
+  template <typename S>
+  using BigPack = ekat::Pack<S,SCREAM_PACK_SIZE>;
+  template <typename S>
+  using SmallPack = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+
+  using Pack = BigPack<Scalar>;
+  using Spack = SmallPack<Scalar>;
+
+  using KT = KokkosTypes<Device>;
+  using MemberType = typename KT::MemberType;
+
+  using WorkspaceManager = typename ekat::WorkspaceManager<Spack, Device>;
+  using Workspace        = typename WorkspaceManager::Workspace;
+
+  template <typename S>
+  using view_1d = typename KT::template view_1d<S>;
+  template <typename S>
+  using view_2d = typename KT::template view_2d<S>;
+  template <typename S>
+  using view_3d = typename KT::template view_3d<S>;
+  template <typename S>
+  using uview_1d = typename ekat::template Unmanaged<view_1d<S> >;
+
+  template <typename S, int N>
+  using view_1d_ptr_array = typename KT::template view_1d_ptr_carray<S, N>;
+
+  /* ------------------------------------------------------------------------------------------- */
+  // SPA structures to help manage all of the variables:
+  struct SPATimeState {
+    SPATimeState() = default;
+    // The current month
+    Int current_month;
+    // Julian Date for the beginning of the month
+    Real t_beg_month;
+    // Current simulation Julian Date
+    Real t_now;
+    // Number of days in the current month, cast as a Real
+    Real days_this_month;
+  }; // SPATimeState
+
+  struct SPAPressureState {
+    SPAPressureState() = default;
+    // Number of vertical levels for the data
```

Comment:
We could use `ncols/nlevs_spa` for the spa data.  My original intent was to have everything stored in a SPA data structure to be spa data and refer to spa data.  So ncols, nlevs, in this context is by definition a reference to SPA data.  In the actual interface I pass the simulation ncols and nlevs.

Reply:
I guess we can leave `ncols` and `nlevs` here, since this struct only stores one resolution. But in places where there are two res handled, we should make an effort to make as clear as possible which is which..